src/errors/kinds.rs

```diff
@@ -31,6 +31,8 @@ pub enum ErrorKind {
     InvalidKey,
     #[strum(message = "Error extracting attribute: {error}")]
     GetAttributeError { error: String },
+    #[strum(message = "The default factory must be a valid zero-argument callable: {error}")]
```

Comment:
This shouldn't be a validation error because it's not a problem in the input data, instead it should be an internal error - e.g. we don't need an `ErrorKind`.

We just need `.map_err(Into::<ValError>::into)?` as per your change in #168.

Reply:
You're completely right. I'll fix this

Reply:
[8b03a91](https://github.com/samuelcolvin/pydantic-core/pull/163/commits/8b03a9121756bb39e651d16e40c32376f8d345df)